### PR TITLE
Update building of Developers' Manual and "Ingesting Data" Section

### PR DESCRIPTION
**Pull request by awholegunch**
**Update: See comment towards bottom, as this PR has changed.**

Remove cdap-file-drop-zone and cdap-file-tailer, as they are no longer supported. 
Removed references to them in the documentation.

Add reference to their removal to the release notes.

See these pages:
- [Ingesting Data](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/developers-manual/ingesting-tools/index.html)
- [Release Notes](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/reference-manual/release-notes.html#general)

### CODE REVIEW

**Comment by awholegunch** [comment]
This needs to be merged before https://github.com/caskdata/cdap-ingest/pull/106, otherwise the documentation build breaks, as it depends on the files in the other repo.

File: cdap-docs/developers-manual/source/index.rst [hunk 1]
```diff
@@ -53,8 +53,7 @@ CDAP Developers’ Manual
 .. _ingesting-tools: ingesting-tools/index.html
 
 - |ingesting-tools|_ CDAP comes with a number of tools to make a developer’s life easier. These
-  tools help with **ingesting data into CDAP:** using Java or Python, the bulk ingestion of files,
-  a daemon to tail local files and an Apache Flume Sink implementation.
+  tools help with **ingesting data into CDAP:** using Java or Python, and an Apache Flume Sink implementation.
```
Comment (anew) [inline cdap-docs/developers-manual/source/index.rst]: we also have ruby and javascript

File: cdap-docs/developers-manual/source/ingesting-tools/index.rst [hunk 2]
```diff
@@ -13,17 +13,13 @@ Ingesting Data into CDAP
...
    cdap-flume
 
 .. rubric:: Introduction
 
 One of the first tasks of actually working with Big Data applications is getting the data in.
 As data ingestion is a fundamental issue, and as one tool often does not fit all needs,
-we have assembled a set of tools and applications to assist in ingesting data into CDAP:
+we have assembled tools and applications to assist in ingesting data into CDAP:
 
 - Java and Python APIs for controlling and writing to Streams;
```
Comment (anew) [inline cdap-docs/developers-manual/source/ingesting-tools/index.rst]: ruby? javascript?

**Comment by anew** [comment]
My main comment is that we have stream clients for javascript and ruby, too. Are they undocumented on purpose?

**Comment by awholegunch** [comment]
New PR is at https://github.com/caskdata/cdap-ingest/pull/107. It has been merged, so I will update this PR to reflect that.

**Comment by awholegunch** [comment]
I have revised this PR heavily. It now adds in the Javascript and Ruby clients, removes unused functions from the build scripts, and have staged a new version. It removes the requirement that Pandoc be installed in order for the documentation to be built.

Pages to view:

- [Ingesting Data into CDAP](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/index.html)
- [File DropZone](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/developers-manual/ingesting-tools/cdap-file-drop-zone.html)
- [File Tailer](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/developers-manual/ingesting-tools/cdap-file-tailer.html)

File: cdap-docs/developers-manual/source/ingesting-tools/cdap-file-drop-zone.rst [hunk 3]
```diff
@@ -4,4 +4,6 @@
 
 .. highlight:: console
 
-.. include:: ../_includes/cdap-file-drop-zone.rst
+.. _cdap-file-drop-zone:
```
Comment (anew) [inline cdap-docs/developers-manual/source/ingesting-tools/cdap-file-drop-zone.rst]: why are you still including that? We don't want to document this any more.

File: cdap-docs/developers-manual/source/ingesting-tools/cdap-file-tailer.rst [hunk 4]
```diff
@@ -4,4 +4,6 @@
 
 .. highlight:: console
 
-.. include:: ../_includes/cdap-file-tailer.rst
+.. _cdap-file-tailer:
```
Comment (anew) [inline cdap-docs/developers-manual/source/ingesting-tools/cdap-file-tailer.rst]: same here

File: cdap-docs/developers-manual/source/ingesting-tools/index.rst [hunk 5]
```diff
@@ -12,18 +12,24 @@ Ingesting Data into CDAP
...
 .. rubric:: Introduction
 
 One of the first tasks of actually working with Big Data applications is getting the data in.
 As data ingestion is a fundamental issue, and as one tool often does not fit all needs,
-we have assembled a set of tools and applications to assist in ingesting data into CDAP:
+we have assembled tools and applications to assist with ingesting data into CDAP:
+
+- Java, JavaScript, Python, and Ruby APIs for controlling and writing to Streams;
+- an Apache Flume Sink implementation for writing events received from a source;
+- a drop zone for bulk ingestion of files; and
```
Comment (anew) [inline cdap-docs/developers-manual/source/ingesting-tools/index.rst]: we do not want to mention these any more

File: cdap-docs/reference-manual/source/release-notes.rst [hunk 6]
```diff
@@ -30,6 +30,10 @@ Cask Data Application Platform Release Notes
 General
 -------
 
+- The :ref:`File DropZone <cdap-file-drop-zone>` and :ref:`File Tailer <cdap-file-tailer>`
+  are no longer supported as of Release 3.0.
+
```
Comment (anew) [inline cdap-docs/reference-manual/source/release-notes.rst]: I think this is the only place where we still need to mention drop zone and file tailer.

**Comment by anew** [comment]
I can't tell what those build scripts do, I trust that you tested them thoroughly. 
Regarding drop zone and file tailer, all we should mention is in the release notes that they are no longer supported.

**Comment by awholegunch** [comment]
Removed references and restaged docs:

Pages to review:
- [Ingesting Data into CDAP](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/developers-manual/ingesting-tools/index.html)
- [Release Notes](http://stg-web101.sw.joyent.continuuity.net/cdap/3.0.0-SNAPSHOT-3.0_edit_dev_man_ingesting/en/reference-manual/release-notes.html#release-3-0-0)

**Comment by anew** [comment]
I am ok with this. The only issue I see is that the JavaScript, Python, and Ruby clients have not been ported to v3 yet, so right now they will not work because v2 APIs have been removed. See CDAP-212[789].

**Comment by awholegunch** [comment]
Are [CDAP-2127](https://issues.cask.co/browse/CDAP-2127),   [CDAP-2128](https://issues.cask.co/browse/CDAP-2128), and  [CDAP-2129](https://issues.cask.co/browse/CDAP-2129) going to be resolved for the 3.0 release?

**Comment by duddun** [comment]
LGTM other than the javascript/python/ruby stream client comments.

**Comment by awholegunch** [comment]
I've removed the Javascript references, and restaged the documentation.

**Comment by anew** [comment]
At the end of the Java Stream Client section, there seems to be a code formatting issue.

**Comment by anew** [comment]
There are formatting issues in the included READMEs from the stream-clients. Probably needs to be fixed separately, because they are in a separate repo.

**Comment by anew** [comment]
The included READMEs from stream-clients have a copyright notice at the end... looks weird here. I guess we do want the copyright in the individual readme's, but not included here. How can that be done?

**Comment by anew** [comment]
There are a couple of issues in the cdap-ingest includes. Since they have to be fixed in a separate PR in that repo, and everything else here looks good: LGTM.

Can you open a Jira for fixing the ingest READMEs to make sure we don't drop the ball on them?

**Comment by awholegunch** [comment]
I might be able to fix it from this end. Let me check.

**Comment by awholegunch** [comment]
I took a look: some I could fix from this end, but I will do it from the other repo.

**Comment by awholegunch** [comment]
This has been updated and restaged with files included from the remote repos.

**Comment by anew** [comment]
There is still a formatting problem in the Python Client, under Creating a StreamWriter.

**Comment by anew** [comment]
The Source Code Repository at the bottom of each section appears either centered or indented. Perhaps it would look better left-flush?

**Comment by anew** [comment]
Other than these two remarks, it LGTM. Please fix and merge without the need for another review.

**Comment by awholegunch** [comment]
Fixed both issues. Merging.

**Pull request merged**